arwen/host/execution.go

```diff
@@ -180,6 +180,20 @@ func (host *vmHost) doRunSmartContractCall(input *vmcommon.ContractCallInput) (v
 	return
 }
 
+func copyTxHashesFromContext(copyEnabled bool, runtime arwen.RuntimeContext, input *vmcommon.ContractCallInput) {
```

Comment:
This could be a method of `runtimeContext`, where the `host` is already available, so `host.IsESDTFunctionsEnabled()` can be called straightforwardly.

Something like `runtime.CopyTxHashesToInput(input)`, everything contained within.

Reply:
it is so particular that I would not make a public function and interface for it.

Reply:
Will eventually move it on the `host` then, as a private method.